index.js

```diff
@@ -3,15 +3,20 @@ var AlexaUtterances = require("alexa-utterances");
 var SSML = require("./to-ssml");
 var alexa = {};
 
-alexa.response = function() {
+alexa.response = function(hasSession) {
```

Comment:
As I see your idea:
```
request.session = {
    this.get = function(key) {...};
    this.set = function(key, value) {...};
    this.clear = function() {...};
    this.details = {...};
    this.attributes = {...};
    this.isNew = bool;
}
```

Just don't understand why we need to store the same data in `this.sessionDetails` and in properties of `this`.

BTW, these changes will break compatibility.

And we need to edit `response.response` from the `session` object. Don't sure how we can do that.